### PR TITLE
LLD: Introduce "Troubleshoot network" panel

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

In context of users working with degraded network conditions due to proxies, VPNs or firewalls, we need an in-house solution to troubleshoot where problems could be. Because of our current LLD architecture, the support of some proxy may only partially work notably because of our "internal process" (and we have plans to improve this, see [internal doc](https://ledgerhq.atlassian.net/wiki/spaces/WALLETCO/pages/3949920483)).

We need however to prove that internal process indeed is the root cause of some situations.

#### Demo of the feature

This PR introduces a "Troubleshoot network" panel that will do simple http calls (and websocket connection) to actively assess the link from the user client to the backend is effectively working. We're assessing that both on the "renderer" (aka the chromium networking) and the "internal" (aka node.js internal process we spawn notably for coins business logic and device connection)

https://user-images.githubusercontent.com/211411/204311525-e5394680-9a4f-43b7-b9d2-a997725b338c.mp4

#### other technical notes

- The troubleshoot network logic was made in live-common and we could go farther by modularising this idea per coin. For now, I only assessed bitcoin and ethereum explorers that way.
- an integration test was also added to non reg on the queries we do :)
- it's overall a very simple mechanism. we could have it on LLM too if we want.

### ❓ Context

- **Impacted projects**: `LLD` 
- **Linked resource(s)**: `N/A` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Review by juan-cortes — APPROVED** [review]

File: libs/ledger-live-common/src/network-troubleshooting/index.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+// Run all checks and return. each troubleshoot have a promise that suceed if the underlying job worked.
+export function troubleshoot(): Troubleshoot[] {
+  // TODO in future, we can delegate a "troubleshoot" per coin implementation.
+  return [
+    {
+      title: "My Ledger services (scriptrunner)",
+      ...websocketConnects(
+        `${getEnv(
+          "BASE_SOCKET_URL"
+        )}/apps/list?targetId=856686596&perso=perso_11&livecommonversion=27.7.2` // TODO have a dummy echo endpoint
```
Comment (chabroA) [inline libs/ledger-live-common/src/network-troubleshooting/index.ts:31]: I see a `livecommonversion=27.7.2` in the query params, will this value need to be updated each time a new LLC version is released?

Comment (gre) [reply libs/ledger-live-common/src/network-troubleshooting/index.ts:31]: for now it's hardcoded, i just need an endpoint that works, it's not really important what this value is, but i think this field may be required (or may not be, not important)

File: apps/ledger-live-desktop/src/renderer/components/TroubleshootNetworkButton.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// @flow
```
Comment (alexandremgo) [inline apps/ledger-live-desktop/src/renderer/components/TroubleshootNetworkButton.jsx:1]: The troubleshooting logic looks good to me, but why still use flow and not typescript ? 🙈

Comment (gre) [reply apps/ledger-live-desktop/src/renderer/components/TroubleshootNetworkButton.jsx:1]: well, this is LLD. we can't do typescript unless all the "imports" are also typescript was my understanding. I cloned an existing component to make these. We will have to work on Typescript LLD as a whole.

File: libs/ledger-live-common/src/network-troubleshooting/index.ts (hunk 1 above)
Comment (gre) [reply libs/ledger-live-common/src/network-troubleshooting/index.ts:31]: (in parallel, i'm asking backend to had such "health" wss version)

**Pull request merged**